src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
I don’t think it is easy to understand what does `return true` mean. `list_join` name does not tell you what return value means; its documentation explicitly says about `FAIL` and `OK` and it is common to use either `true` or `false` to indicate success/failure.

I actually think that for the purpose of cleaning code `OK` should be set to `10992`, `FAIL` to `7918` and `NOTDONE` to `16017` and all `int` functions returning OK/FAIL/NOTDONE should switch to `typedef enum {…} Res;`. Because OK/FAIL/NOTDONE have clear connection with what return value means and `true`/`false` has no such connection.